<h4 align="center" style="height:1123px;font-family: 'Times New Roman'"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “OPERATOR PADA PYTHON”

    Disusun Oleh :

    Rhakez Elvano Stonie

    2611532013

    Dosen Pengampu:

    Wahyudi.DR.. S.T.M.T

    Asisten Praktikum:

    Rifky Yuliandra

<br>

 <img src="../images/logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI

    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</b></h4>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">

<h2 align="center">KATA PENGANTAR</h2>
Puji syukur kehadirat Tuhan Yang Maha Esa karena atas rahmat dan karunia-Nya laporan praktikum yang berjudul “Operator pada Python” ini dapat diselesaikan dengan baik. Laporan ini disusun untuk memenuhi salah satu tugas praktikum serta membantu memahami penggunaan operator dalam bahasa pemrograman Python.

Dalam praktikum ini, penulis mempelajari berbagai jenis operator yang terdapat pada Python, seperti operator aritmatika, operator perbandingan, operator logika, operator penugasan, operator identitas, operator keanggotaan, dan operator bitwise. Melalui praktikum ini, penulis diharapkan dapat memahami fungsi dari setiap operator serta mampu menerapkannya dalam pembuatan program sederhana.

Penulis menyadari bahwa laporan ini masih memiliki kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan agar laporan berikutnya dapat menjadi lebih baik. Semoga laporan ini dapat memberikan manfaat bagi penulis maupun pembaca dalam memahami penggunaan operator pada bahasa pemrograman Python.


<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
Rhakez Elvano Stonie
<br>
2611532013
</div>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB I<br>PENDAHULUAN</h1>

## 1.1 Latar Belakang
Python merupakan salah satu bahasa pemrograman yang banyak digunakan karena memiliki sintaks yang sederhana dan mudah dipahami. Dalam pembuatan program, operator menjadi salah satu bagian penting yang digunakan untuk melakukan berbagai proses terhadap data. Operator pada Python dapat digunakan untuk melakukan perhitungan, membandingkan nilai, menggabungkan kondisi, memberikan nilai pada variabel, serta melakukan operasi lainnya. Jenis operator yang terdapat dalam Python antara lain operator aritmatika, perbandingan, logika, penugasan, identitas, keanggotaan, dan bitwise. Oleh karena itu, pemahaman mengenai operator sangat diperlukan agar dapat membuat program yang sesuai dengan kebutuhan dan menghasilkan proses yang benar. Melalui praktikum ini, mahasiswa diharapkan dapat memahami fungsi dari berbagai operator pada Python serta mampu menerapkannya dalam program sederhana.

## 1.2 Tujuan
1. Mengetahui pengertian dan fungsi operator pada bahasa pemrograman Python.
2. Memahami berbagai jenis operator pada Python, seperti operator aritmatika, perbandingan, logika, penugasan, identitas, keanggotaan, dan bitwise.
3. Mengetahui cara penggunaan setiap jenis operator dalam program Python.
4. Mampu menerapkan operator pada Python untuk melakukan perhitungan dan pengolahan data.
5. Meningkatkan kemampuan dalam membuat program sederhana menggunakan operator Python.

## 1.3 Manfaat
1. Menambah pemahaman mengenai konsep dan fungsi operator pada bahasa pemrograman Python.
2. Membantu memahami penggunaan berbagai jenis operator dalam pengolahan data.
3. Meningkatkan kemampuan dalam menerapkan operator untuk membuat program sederhana.
4. Melatih kemampuan berpikir logis dan sistematis dalam menyelesaikan masalah menggunakan Python.
5. Menjadi dasar untuk mempelajari materi pemrograman Python yang lebih lanjut.
</div>

---
<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB II<br>PEMBAHASAN</h1>

## 2.1 Dasar Teori
Python memiliki berbagai jenis operator yang digunakan untuk melakukan proses tertentu terhadap data dan variabel. Operator merupakan simbol atau kata khusus yang digunakan untuk melakukan operasi, seperti perhitungan, perbandingan, pemberian nilai, maupun pengolahan kondisi. Penggunaan operator menjadi salah satu dasar penting dalam pemrograman karena hampir setiap program membutuhkan operasi terhadap data.

1. Operator Aritmatika
Operator aritmatika digunakan untuk melakukan operasi perhitungan matematika pada Python. Operator yang termasuk di dalamnya adalah penjumlahan (+), pengurangan (-), perkalian (*), pembagian (/), pembagian bulat (//), modulus atau sisa pembagian (%), dan perpangkatan (**). Operator aritmatika dapat digunakan pada berbagai tipe data numerik, seperti integer dan float. Penggunaannya banyak ditemukan dalam perhitungan nilai, transaksi, luas bangun, dan berbagai proses perhitungan lainnya.
2. Operator Perbandingan
Operator perbandingan digunakan untuk membandingkan dua nilai atau lebih. Hasil dari operasi perbandingan berupa nilai Boolean, yaitu True atau False. Operator yang digunakan meliputi sama dengan (==), tidak sama dengan (!=), lebih besar (>), lebih kecil (<), lebih besar atau sama dengan (>=), serta lebih kecil atau sama dengan (<=). Operator ini biasanya digunakan dalam percabangan dan pemeriksaan kondisi pada suatu program.
3. Operator Logika
Operator logika digunakan untuk menggabungkan atau memanipulasi beberapa kondisi. Python memiliki tiga operator logika utama, yaitu and, or, dan not. Operator and menghasilkan nilai benar apabila semua kondisi yang diperiksa bernilai benar, sedangkan or menghasilkan nilai benar apabila terdapat setidaknya satu kondisi yang benar. Sementara itu, not digunakan untuk membalik nilai Boolean dari suatu kondisi. Operator logika banyak digunakan dalam proses pengambilan keputusan pada program.
4. Operator Penugasan
Operator penugasan digunakan untuk memberikan nilai kepada variabel atau mengubah nilai yang telah tersimpan di dalam variabel. Operator yang paling dasar adalah =, sedangkan operator gabungan lainnya antara lain +=, -=, *=, /=, //=, %= dan **=. Operator tersebut dapat membuat penulisan program menjadi lebih sederhana, terutama ketika suatu variabel perlu diperbarui berdasarkan nilai sebelumnya.
5. Operator Bitwise
Operator bitwise digunakan untuk melakukan operasi pada tingkat bit dari suatu bilangan integer. Python menyediakan beberapa operator bitwise, yaitu AND (&), OR (|), XOR (^), NOT (~), left shift (<<), dan right shift (>>). Operasi ini bekerja dengan memproses representasi biner dari suatu bilangan. Operator bitwise biasanya digunakan dalam pengolahan data tingkat rendah, manipulasi bit, sistem komputer, dan beberapa kebutuhan pemrograman tertentu.
6. Operator Keanggotaan
Operator keanggotaan digunakan untuk memeriksa apakah suatu nilai terdapat di dalam sebuah objek seperti string, list, tuple, atau set. Python memiliki operator in dan not in. Operator in menghasilkan True apabila suatu nilai ditemukan di dalam objek, sedangkan not in menghasilkan True apabila nilai tersebut tidak ditemukan. Operator ini dapat digunakan untuk melakukan pemeriksaan data dengan cara yang sederhana.
7. Operator Identitas
Operator identitas digunakan untuk mengetahui apakah dua variabel mengacu pada objek yang sama di dalam memori. Python menyediakan operator is dan is not. Operator is menghasilkan True apabila kedua variabel mengacu pada objek yang sama, sedangkan is not menghasilkan True apabila keduanya tidak mengacu pada objek yang sama. Operator identitas berbeda dengan operator == yang digunakan untuk membandingkan nilai.

## 2.2 Kode Program

### 2.2.1 Program aritmatika_2611532013
</div>

In [1]:
# Buat file dengan nama aritmatika_NIM.py
# Buat program untuk operator aritmatika dalam Python
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer

angka1_2013 = int(input("Input angka-1: "))
angka2_2013 = int(input("Input angka-2: "))

# Penjumlahan
hasil_2013 = angka1_2013 + angka2_2013
print("\nOperator Penjumlahan")
print("Hasil =",hasil_2013)

# Pengurangan
hasil_2013 = angka1_2013 - angka2_2013
print("\nOperator Pengurangan")
print("Hasil =",hasil_2013)

# Perkalian
hasil_2013 = angka1_2013 * angka2_2013
print("\nOperator Perkalian")
print("Hasil =",hasil_2013)

# Pembagian, pembagian bulat, dan sisa bagi
if angka2_2013 != 0:
    hasil_2013 = angka1_2013 / angka2_2013
    print("\nOperator Pembagian")
    print("Hasil =",hasil_2013)

    hasil_2013 = angka1_2013 // angka2_2013
    print("\nOperator Pembagian Bulat")
    print("Hasil =",hasil_2013)

    hasil_2013 = angka1_2013 % angka2_2013
    print("\nOperator Sisa Bagi")
    print("Hasil =",hasil_2013)
else:
    print("Angka kedua tidak boleh bernilai 0.")

# Pangkat
hasil_2013 = angka1_2013 ** angka2_2013
print("\nOperator Pangkat")
print("Hasil =",hasil_2013)


Operator Penjumlahan
Hasil = 12

Operator Pengurangan
Hasil = -2

Operator Perkalian
Hasil = 35

Operator Pembagian
Hasil = 0.7142857142857143

Operator Pembagian Bulat
Hasil = 0

Operator Sisa Bagi
Hasil = 5

Operator Pangkat
Hasil = 78125


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_2013 = int(input("Input angka-1: "))<br>
angka2_2013 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima data, sedangkan int() mengubah data yang dimasukkan menjadi tipe integer.
2. Operator Penjumlahan<br>
hasil_2013 = angka1_2013 + angka2_2013<br>
Operator + digunakan untuk menjumlahkan angka1_2013 dengan angka2_2013, kemudian hasilnya disimpan ke variabel hasil_2013.
3. Operator Pengurangan<br>
hasil_2013 = angka1_2013 - angka2_2013<br>
Operator - digunakan untuk mengurangi nilai angka1_2013 dengan angka2_2013.
4. Operator Perkalian<br>
hasil_2013 = angka1_2013 * angka2_2013<br>
Operator * digunakan untuk mengalikan kedua angka yang telah dimasukkan.
5. Operator Pembagian<br>
hasil_2013 = angka1_2013 / angka2_2013<br>
Operator / digunakan untuk melakukan pembagian. Hasil pembagian biasanya berupa tipe data float.
6. Operator Pembagian Bulat<br>
hasil_2013 = angka1_2013 // angka2_2013<br>
Operator // digunakan untuk melakukan pembagian dan mengambil hasil bilangan bulatnya saja, tanpa bagian desimal.
7. Operator Sisa Bagi<br>
hasil_2013 = angka1_2013 % angka2_2013<br>
Operator % digunakan untuk mendapatkan sisa dari hasil pembagian antara angka1_2013 dan angka2_2013.
8. Pengecekan angka kedua<br>
if angka2_2013 != 0:<br>
Program memeriksa apakah angka kedua tidak bernilai 0. Hal ini dilakukan karena pembagian dengan angka 0 tidak diperbolehkan.<br>
Jika angka2_2013 bernilai 0, program menjalankan:<br>
print("Angka kedua tidak boleh bernilai 0.")
9. Operator Pangkat<br>
hasil_2013 = angka1_2013 ** angka2_2013<br>
Operator ** digunakan untuk menghitung pangkat, yaitu angka1_2013 dipangkatkan dengan angka2_2013.


### 2.2.2 Program assigment_2611532013
</div>

In [2]:
# Buat file dengan nama assigment_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator assigment dalam Python

angka1_2013 = int(input("Input angka-1: "))
angka2_2013 = int(input("Input angka-2: "))

print("\nNilai awal angka1 =",angka1_2013)
print("Nilai angka2 =",angka2_2013)

# Assigment biasa
hasil_2013 = angka1_2013 
print("\nAssigment biasa (=)")
print("Hasil =",hasil_2013)

# Assigment penambahan
hasil_2013 = angka1_2013 
hasil_2013 += angka2_2013 
print("\nAssigment penambahan (+=)")
print("Hasil =",hasil_2013)

# Assigment pengurangan
hasil_2013 = angka1_2013 
hasil_2013 -= angka2_2013 
print("\nAssigment pengurangan (-=)")
print("Hasil =",hasil_2013)

# Assigment perkalian
hasil_2013 = angka1_2013 
hasil_2013 *= angka2_2013
print("\nAssigment perkalian (*=)")
print("Hasil =",hasil_2013)

# Assigment pembagian, pembagian bulat, dan sisa bagi
if angka2_2013 != 0:
    hasil_2013 = angka1_2013 
    hasil_2013 /= angka2_2013
    print("\nAssigment pembagian (/=)")
    print("Hasil =",hasil_2013)
    # Operator tambahan
    hasil_2013 = angka1_2013 
    hasil_2013 //= angka2_2013
    print("\nAssigment pembagian bulat (//=)")
    print("Hasil =",hasil_2013)
    hasil_2013 = angka1_2013 
    hasil_2013 %= angka2_2013
    print("\nOperator sisa bagi (%=)")
    print("Hasil =",hasil_2013)
else:
    print("\nPembagian tidak dapat dilakukan.")
    print("Angka kedua tidak boleh bernilai 0.")

# Operator tambahan: assigment perpangkatan
hasil_2013 = angka1_2013 
hasil_2013 **= angka2_2013
print("\nOperator perpangkatan (**=)")
print("Hasil =",hasil_2013)


Nilai awal angka1 = 5
Nilai angka2 = 5

Assigment biasa (=)
Hasil = 5

Assigment penambahan (+=)
Hasil = 10

Assigment pengurangan (-=)
Hasil = 0

Assigment perkalian (*=)
Hasil = 25

Assigment pembagian (/=)
Hasil = 1.0

Assigment pembagian bulat (//=)
Hasil = 1

Operator sisa bagi (%=)
Hasil = 0

Operator perpangkatan (**=)
Hasil = 3125


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_2013 = int(input("Input angka-1: "))<br>
angka2_2013 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan tersebut menjadi tipe data integer.
2. Menampilkan nilai awal<br>
print("\nNilai awal angka1 =", angka1_2013)<br>
print("Nilai angka2 =", angka2_2013)<br>
Bagian ini digunakan untuk menampilkan nilai awal dari angka1_2013 dan angka2_2013.
3. Assignment biasa (=)<br>
hasil_2013 = angka1_2013<br>
Operator = digunakan untuk memberikan atau menetapkan nilai dari angka1_2013 ke variabel hasil_2013.
4. Assignment penambahan (+=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 += angka2_2013<br>
Operator += digunakan untuk menambahkan nilai angka2_2013 ke hasil_2013.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 + angka2_2013
5. Assignment pengurangan (-=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 -= angka2_2013<br>
Operator -= digunakan untuk mengurangi hasil_2013 dengan nilai angka2_2013.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 - angka2_2013
6. Assignment perkalian (*=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 *= angka2_2013<br>
Operator *= digunakan untuk mengalikan hasil_2013 dengan angka2_2013.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 * angka2_2013
7. Assignment pembagian (/=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 /= angka2_2013<br>
Operator /= digunakan untuk membagi nilai hasil_2013 dengan angka2_2013.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 / angka2_2013<br>
Hasil pembagian menggunakan / biasanya berupa float.
8. Assignment pembagian bulat (//=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 //= angka2_2013<br>
Operator //= digunakan untuk melakukan pembagian bulat.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 // angka2_2013
9. Assignment sisa bagi (%=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 %= angka2_2013<br>
Operator %= digunakan untuk mendapatkan sisa hasil pembagian.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 % angka2_2013
10. Pengecekan nilai 0<br>
if angka2_2013 != 0:<br>
Program memeriksa apakah angka2_2013 tidak sama dengan 0. Pemeriksaan ini diperlukan karena operasi pembagian, pembagian bulat, dan sisa bagi tidak dapat dilakukan dengan pembagi 0.<br>
Jika nilainya 0, program menampilkan:<br>
print("\nPembagian tidak dapat dilakukan.")<br>
print("Angka kedua tidak boleh bernilai 0.")
11. Assignment perpangkatan (**=)<br>
hasil_2013 = angka1_2013<br>
hasil_2013 **= angka2_2013<br>
Operator **= digunakan untuk melakukan perpangkatan.<br>
Bentuk sederhananya:<br>
hasil_2013 = hasil_2013 ** angka2_2013
### 2.2.3 Program perbandingan_2611532013
</div>

In [5]:
# Buat file dengan nama perbandingan_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator perbandingan dalam Python

angka1_2013 = int(input("Input angka-1:"))
angka2_2013 = int(input("Input angka-2:"))

# Lebih besar dari
hasil_2013 = angka1_2013 > angka2_2013
print("\nOperator lebih besar dari")
print("angka1_2013 > angka2_2013 =", hasil_2013)

# Lebih kecil dari
hasil_2013 = angka1_2013 < angka2_2013
print("\nOperator lebih kecil dari")
print("angka1_2013 < angka2_2013 =", hasil_2013)

# Lebih besar dari atau sama dengan
hasil_2013 = angka1_2013 >= angka2_2013
print("\nOperator lebih besar dari atau sama dengan")
print("angka1_2013 >= angka2_2013 =", hasil_2013)

# Lebih kecil dari atau sama dengan
hasil_2013 = angka1_2013 <= angka2_2013
print("\nOperator lebih kecil dari atau sama dengan")
print("angka1_2013 <= angka2_2013 =", hasil_2013)

# Sama dengan
hasil_2013 = angka1_2013 == angka2_2013
print("\nOperator sama dengan")
print("angka1_2013 == angka2_2013 =", hasil_2013)

# Tidak sama dengan
hasil_2013 = angka1_2013 != angka2_2013
print("\nOperator tidak sama dengan")
print("angka1_2013 != angka2_2013 =", hasil_2013)

# Tambahan: Perbandingan berantai dalam python
hasil_2013 = 0 < angka1_2013 < 100
print("\nPerbandingan berantai")
print("0 < angka1_2013 < 100 =", hasil_2013)

hasil_2013 = 0 < angka2_2013 < 100
print("0 < angka2_2013 < 100 =", hasil_2013)


Operator lebih besar dari
angka1_2013 > angka2_2013 = True

Operator lebih kecil dari
angka1_2013 < angka2_2013 = False

Operator lebih besar dari atau sama dengan
angka1_2013 >= angka2_2013 = True

Operator lebih kecil dari atau sama dengan
angka1_2013 <= angka2_2013 = False

Operator sama dengan
angka1_2013 == angka2_2013 = False

Operator tidak sama dengan
angka1_2013 != angka2_2013 = True

Perbandingan berantai
0 < angka1_2013 < 100 = True
0 < angka2_2013 < 100 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_2013 = int(input("Input angka-1: "))<br>
angka2_2013 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
2. Operator lebih besar dari (>)<br>
hasil_2013 = angka1_2013 > angka2_2013<br>
Operator > digunakan untuk mengecek apakah angka1_2013 lebih besar daripada angka2_2013.<br>
Hasilnya berupa nilai Boolean, yaitu True atau False.
3. Operator lebih kecil dari (<)<br>
hasil_2013 = angka1_2013 < angka2_2013<br>
Operator < digunakan untuk mengecek apakah angka1_2013 lebih kecil daripada angka2_2013.
4. Operator lebih besar dari atau sama dengan (>=)<br>
hasil_2013 = angka1_2013 >= angka2_2013<br>
Operator >= mengecek apakah angka1_2013 lebih besar atau sama dengan angka2_2013.
5. Operator lebih kecil dari atau sama dengan (<=)<br>
hasil_2013 = angka1_2013 <= angka2_2013<br>
Operator <= mengecek apakah angka1_2013 lebih kecil atau sama dengan angka2_2013.
6. Operator sama dengan (==)<br>
hasil_2013 = angka1_2013 == angka2_2013<br>
Operator == digunakan untuk mengecek apakah kedua angka memiliki nilai yang sama.<br>
Perlu dibedakan antara = dan ==.<br>
= digunakan untuk memberikan nilai, sedangkan == digunakan untuk membandingkan nilai.
7. Operator tidak sama dengan (!=)<br>
hasil_2013 = angka1_2013 != angka2_2013<br>
Operator != digunakan untuk mengecek apakah angka1_2013 dan angka2_2013 memiliki nilai yang berbeda.
8. Perbandingan berantai untuk angka pertama<br>
hasil_2013 = 0 < angka1_2013 < 100<br>
Kode tersebut merupakan perbandingan berantai (chained comparison). Artinya, program mengecek apakah angka1_2013 berada di antara 0 dan 100, dengan syarat:<br>
0 < angka1_2013 dan angka1_2013 < 100<br>
Jika kedua kondisi benar, hasilnya True.
9. Perbandingan berantai untuk angka kedua<br>
hasil_2013 = 0 < angka2_2013 < 100<br>
Bagian ini memiliki fungsi yang sama, tetapi digunakan untuk mengecek apakah angka2_2013 berada di antara 0 dan 100.

### 2.2.4 Program logika_2611532013
</div>

In [6]:
# Buat file dengan nama logika_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: a1_1234
# Program ini menngunakan fungsi input()
# Program operator logika dalam Python

# Memasukan nilai boolean
# Input tidak peka terhadap huruf besar dan kecil
a1_2013 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_2013 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"

print("\nA1 = ",a1_2013)
print("A2 = ",a2_2013)

# Konjungsi: bernilai True jika keduanya True
hasil_2013 = a1_2013 and a2_2013
print("\nKonjungsi (AND)")
print("A1 and A2 =",hasil_2013)

# Disjungsi: bernilai True jika salah satunya False
hasil_2013 = a1_2013 or a2_2013
print("\nDisjungsi (OR)")
print("A1 or A2 =",hasil_2013)

# Negasi A1: membalik nilai A1
hasil_2013 = not a1_2013
print("\nNegasi A1(NOT)")
print("not A1",hasil_2013)

# Negasi A2: membalik nilai A2
hasil_2013 = not a2_2013
print("\nNegasi A2(NOT)")
print("not A2",hasil_2013)

# XOR: bernilai True jika kedua nilai berbeda
hasil_2013 = a1_2013 != a2_2013
print("\nDisjungsi Ekslusif (XOR)")
print("A1 XOR A2",hasil_2013)


A1 =  False
A2 =  False

Konjungsi (AND)
A1 and A2 = False

Disjungsi (OR)
A1 or A2 = False

Negasi A1(NOT)
not A1 True

Negasi A2(NOT)
not A2 True

Disjungsi Ekslusif (XOR)
A1 XOR A2 False


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Memasukkan nilai Boolean<br>
a1_2013 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"<br>
a2_2013 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"<br>
Program meminta pengguna memasukkan nilai true atau false.<br>
input() menerima masukan dari pengguna.<br>
.strip() menghilangkan spasi di awal dan akhir.<br>
.lower() mengubah huruf menjadi huruf kecil sehingga TRUE, True, dan true dapat diproses dengan cara yang sama.<br>
== "true" membandingkan masukan dengan teks "true" dan menghasilkan nilai Boolean True atau False.
2. Menampilkan nilai Boolean<br>
print("\nA1 = ", a1_2013)<br>
print("A2 = ", a2_2013)<br>
Bagian ini menampilkan nilai Boolean yang telah dimasukkan dan dikonversi oleh program.
3. Konjungsi (AND)
hasil_2013 = a1_2013 and a2_2013<br>
Operator and digunakan untuk menggabungkan dua kondisi. Hasilnya akan True hanya jika kedua nilai bernilai True.<br>
Contoh:<br>
A1 = True<br>
A2 = True<br>
Hasil = True<br>
Jika salah satu atau keduanya False, hasilnya False.
4. Disjungsi (OR)<br>
hasil_2013 = a1_2013 or a2_2013<br>
Operator or menghasilkan True jika minimal salah satu nilai bernilai True.<br>
Hasilnya hanya False jika kedua nilai bernilai False.
5. Negasi A1 (NOT)<br>
hasil_2013 = not a1_2013<br>
Operator not digunakan untuk membalik nilai Boolean dari a1_2013.<br>
Jika A1 = True, maka not A1 = False.<br>
Jika A1 = False, maka not A1 = True.
6. Negasi A2 (NOT)<br>
hasil_2013 = not a2_2013<br>
Bagian ini memiliki fungsi yang sama dengan sebelumnya, tetapi digunakan untuk membalik nilai Boolean a2_2013.
7. Disjungsi Eksklusif (XOR)<br>
hasil_2013 = a1_2013 != a2_2013<br>
Operator XOR digunakan untuk menghasilkan True jika kedua nilai berbeda.

### 2.2.5 Program lainnya_2611532013
</div>

In [11]:
# Buat file dengan nama lainnya_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()
# Program operator keanggotaan dan identitas

print("======================================")
print("1. OPERATOR KEANGGOTAAN")
print("======================================")

# Input beberapa data yang dipisahkan dengan koma
input_data_2013 = input("Masukkan beberapa angka, pisahkan dengan koma: ")

# Mengubah input menjadi list integer
data_2013 = [int(angka.strip()) for angka in input_data_2013.split(',')]

nilai_dicari_2013 = int(input("Masukkan angka yang ingin dicari: "))

# Operator in
hasil_2013 = nilai_dicari_2013 in data_2013
print("\nOperator keanggotaan IN")
print(nilai_dicari_2013,"in",data_2013,"=",hasil_2013)

# Operator not in
hasil_2013 = nilai_dicari_2013 not in data_2013
print("\nOperator keanggotaan NOT IN")
print(nilai_dicari_2013,"not in",data_2013,"=",hasil_2013)

print("======================================")
print("2. OPERATOR IDENTIAS")
print("======================================")

# objek1 menggunakan list dari input perngguna
object1_2013 = data_2013

# objek2 merujuk pada objek yang sama dengan objek1
object2_2013 = object1_2013

# objek3 memiliki isi sama, tetapi merupakan objek baru
object3_2013 = data_2013.copy()

print("object1 =",object1_2013)
print("object2 =",object2_2013)
print("object3 =",object3_2013)

# Operator is
hasil_2013 = object1_2013 is object2_2013
print("\nOperator identitas IS")
print("objek1 is objek2 =",hasil_2013)

# Operator is not
hasil_2013 = object1_2013 is not object2_2013
print("\nOperator identitas IS NOT")
print("objek1 is not objek2 =",hasil_2013)

# Membandingkan identitas dan nilai
print("\nPerbandingan identitas dan nilai")
print("objek1 is objek3 =",object1_2013 is object3_2013)
print("objek1 == objek3 =",object1_2013 == object3_2013)

1. OPERATOR KEANGGOTAAN

Operator keanggotaan IN
67 in [89, 68, 98, 76, 45] = False

Operator keanggotaan NOT IN
67 not in [89, 68, 98, 76, 45] = True
2. OPERATOR IDENTIAS
object1 = [89, 68, 98, 76, 45]
object2 = [89, 68, 98, 76, 45]
object3 = [89, 68, 98, 76, 45]

Operator identitas IS
objek1 is objek2 = True

Operator identitas IS NOT
objek1 is not objek2 = False

Perbandingan identitas dan nilai
objek1 is objek3 = False
objek1 == objek3 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul program<br>
print("======================================")<br>
print("1. OPERATOR KEANGGOTAAN")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian Operator Keanggotaan agar output program lebih terstruktur.
2. Menerima input beberapa angka
input_data_2013 = input("Masukkan beberapa angka, pisahkan dengan koma: ")<br>
Program meminta pengguna memasukkan beberapa angka yang dipisahkan dengan tanda koma. Input tersebut awalnya disimpan sebagai string.<br>
Contoh:<br>
10, 20, 30, 40
3. Mengubah input menjadi list integer<br>
data_2013 = [int(angka.strip()) for angka in input_data_2013.split(',')]<br>
Kode ini mengubah input string menjadi list yang berisi integer.<br>
.split(',') memisahkan data berdasarkan tanda koma.<br>
.strip() menghilangkan spasi yang tidak diperlukan.<br>
int() mengubah setiap data menjadi integer.<br>
[ ... for ... ] digunakan untuk membuat list.<br>
Contoh:<br>
Input: 10, 20, 30<br>
Hasil: [10, 20, 30]
4. Memasukkan nilai yang ingin dicari<br>
nilai_dicari_2013 = int(input("Masukkan angka yang ingin dicari: "))<br>
Program meminta pengguna memasukkan angka yang akan dicari di dalam data_2013. Input dikonversi menjadi tipe data integer.
5. Operator keanggotaan in<br>
hasil_2013 = nilai_dicari_2013 in data_2013<br>
Operator in digunakan untuk mengecek apakah suatu nilai terdapat di dalam sebuah list.<br>
Jika nilai ditemukan, hasilnya True. Jika tidak ditemukan, hasilnya False.<br>
Contoh:<br>
data = [10, 20, 30]<br>
20 in data → True<br>
50 in data → False
6. Operator keanggotaan not in<br>
hasil_2013 = nilai_dicari_2013 not in data_2013<br>
Operator not in digunakan untuk mengecek apakah suatu nilai tidak terdapat di dalam list.<br>
Jika nilai tidak ditemukan, hasilnya True. Jika ditemukan, hasilnya False.
7. Menampilkan judul operator identitas<br>
print("======================================")<br>
print("2. OPERATOR IDENTIAS")<br>
print("======================================")<br>
Bagian ini menampilkan judul untuk bagian Operator Identitas.
8. Membuat object1<br>
object1_2013 = data_2013<br>
Variabel object1_2013 menunjuk ke objek list yang sama dengan data_2013.
9. Membuat object2<br>
object2_2013 = object1_2013<br>
Variabel object2_2013 menunjuk ke objek yang sama dengan object1_2013.<br>
Artinya, object1_2013 dan object2_2013 memiliki identitas objek yang sama.
10. Membuat object3<br>
object3_2013 = data_2013.copy()<br>
copy() membuat objek list baru yang memiliki isi yang sama dengan data_2013.<br>
Jadi, nilai object3_2013 sama dengan object1_2013, tetapi objeknya berbeda.
11. Menampilkan isi objek<br>
print("object1 =",object1_2013)<br>
print("object2 =",object2_2013)<br>
print("object3 =",object3_2013)<br>
Digunakan untuk menampilkan isi dari ketiga objek.<br>
Karena ketiganya memiliki isi yang sama, hasil yang ditampilkan akan terlihat sama.
12. Operator identitas is<br>
hasil_2013 = object1_2013 is object2_2013<br>
Operator is digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang sama.<br>
Karena object2_2013 dibuat dari object1_2013, hasilnya adalah:<br>
True
13. Operator identitas is not<br>
hasil_2013 = object1_2013 is not object2_2013<br>
Operator is not digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang berbeda.<br>
Karena object1_2013 dan object2_2013 menunjuk objek yang sama, hasilnya adalah:<br>
False
14. Membandingkan identitas dan nilai<br>
print("objek1 is objek3 =",object1_2013 is object3_2013)<br>
print("objek1 == objek3 =",object1_2013 == object3_2013)<br>
Baris pertama menggunakan is untuk membandingkan identitas objek. Karena object3_2013 dibuat menggunakan .copy(), objeknya berbeda sehingga hasilnya:<br>
False<br>
Baris kedua menggunakan == untuk membandingkan nilai atau isi objek. Karena isi object1_2013 dan object3_2013 sama, hasilnya:<br>
True

### 2.2.6 Program bitwise_2611532013
</div>

In [ ]:
# Buat file dengan nama bitwise_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()

print("======================================")
print("3. OPERATOR BITWISE")
print("======================================")

angka1_2013 = int(input("Masukkan angka bitwise-1: "))
angka2_2013 = int(input("Masukkan angka bitwise-2: "))

print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =",angka1_2013,"| biner =",bin(angka1_2013))
print("angka1 =",angka2_2013,"| biner =",bin(angka2_2013))

# Bitwise AND
hasil_2013 = angka1_2013 & angka2_2013
print("\nBitwise AND (&)")
print(angka1_2013,"&",angka2_2013,hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

# Bitwise OR
hasil_2013 = angka1_2013 | angka2_2013
print("\nBitwise OR (|)")
print(angka1_2013,"|",angka2_2013,hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

# Bitwise XOR
hasil_2013 = angka1_2013 ^ angka2_2013
print("\nBitwise XOR (^)")
print(angka1_2013,"^",angka2_2013,hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

# Bitwise NOT
hasil_2013 = ~angka1_2013
print("\nBitwise NOT (~)")
print(angka1_2013,"~",angka2_2013,hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

# Bitwise geser kiri
jumlah_geser_2013 = int(input("\nMasukkan jumlah pergeseran bit: "))

hasil_2013 = angka1_2013 << jumlah_geser_2013
print("\nBitwise geser kiri (<<)")
print(angka1_2013,"<<",jumlah_geser_2013,"=",hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

# Bitwise geser kanan
hasil_2013 = angka1_2013 >> jumlah_geser_2013
print("\nBitwise geser kanan (>>)")
print(angka1_2013,">>",jumlah_geser_2013,"=",hasil_2013)
print("Biner hasil =",bin(hasil_2013))
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

3. OPERATOR BITWISE

Angka dalam bentuk desimal dan biner
angka1 = 10 | biner = 0b1010
angka1 = 20 | biner = 0b10100

Bitwise AND (&)
10 & 20 0
Biner hasil = 0b0
Biner hasil (8 bit) = 00000000

Bitwise OR (|)
10 | 20 30
Biner hasil = 0b11110
Biner hasil (8 bit) = 00011110

Bitwise XOR (^)
10 ^ 20 30
Biner hasil = 0b11110
Biner hasil (8 bit) = 00011110

Bitwise NOT (~)
10 ~ 20 -11
Biner hasil = -0b1011
Biner hasil (8 bit) = -0001011

Bitwise geser kiri (<<)
10 << 1 = 20
Biner hasil = 0b10100
Biner hasil (8 bit) = 00010100

Bitwise geser kanan (>>)
10 >> 1 = 5
Biner hasil = 0b101
Biner hasil (8 bit) = 00000101


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul<br>
print("======================================")<br>
print("3. OPERATOR BITWISE")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian program agar output terlihat lebih terstruktur.
2. Memasukkan dua angka<br>
angka1_2013 = int(input("Masukkan angka bitwise-1: "))<br>
angka2_2013 = int(input("Masukkan angka bitwise-2: "))<br>
Program meminta pengguna memasukkan dua angka. input() menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
3. Menampilkan angka desimal dan biner<br>
print("\nAngka dalam bentuk desimal dan biner")<br>
print("angka1 =",angka1_2013,"| biner =",bin(angka1_2013))<br>
print("angka1 =",angka2_2013,"| biner =",bin(angka2_2013))<br>
Menampilkan nilai angka dalam bentuk desimal dan biner. Fungsi bin() digunakan untuk mengubah bilangan integer menjadi bentuk biner.
4. Operasi AND (&)<br>
hasil_2013 = angka1_2013 & angka2_2013<br>
Operator & membandingkan setiap bit dari kedua angka. Hasil bit akan 1 jika kedua bit bernilai 1.<br>
print(angka1_2013,"&",angka2_2013,hasil_2013)<br>
print("Biner hasil =",bin(hasil_2013))<br>
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))<br>
Menampilkan hasil dalam bentuk desimal, biner biasa, dan biner dengan panjang 8 bit. format(..., "08b") digunakan untuk menampilkan biner minimal 8 digit.
5. Operasi OR (|)<br>
hasil_2013 = angka1_2013 | angka2_2013<br>
Operator | menghasilkan bit 1 jika salah satu atau kedua bit bernilai 1.<br>
print(angka1_2013,"|",angka2_2013,hasil_2013)<br>
print("Biner hasil =",bin(hasil_2013))<br>
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))<br>
Menampilkan hasil operasi dalam bentuk desimal dan biner.
6. Operasi XOR (^)<br>
hasil_2013 = angka1_2013 ^ angka2_2013<br>
Operator ^ menghasilkan bit 1 jika kedua bit berbeda.<br>
Contoh:<br>
0 ^ 0 = 0<br>
0 ^ 1 = 1<br>
1 ^ 0 = 1<br>
1 ^ 1 = 0<br>
Bagian print() berikutnya menampilkan hasil dalam bentuk desimal dan biner.
7. Operasi NOT (~)<br>
hasil_2013 = ~angka1_2013<br>
Operator ~ digunakan untuk membalik bit dari angka1_2013. Bit 1 menjadi 0 dan bit 0 menjadi 1.<br>
print(angka1_2013,"~",angka2_2013,hasil_2013)<br>
Menampilkan hasil operasi NOT.<br>
Namun, pada baris tersebut angka2_2013 ikut ditampilkan meskipun operasi NOT hanya dilakukan terhadap angka1_2013. Jika ingin output lebih sesuai, sebaiknya:<br>
print("~", angka1_2013, "=", hasil_2013)
8. Memasukkan jumlah pergeseran bit<br>
jumlah_geser_2013 = int(input("\nMasukkan jumlah pergeseran bit: "))<br>
Program meminta jumlah posisi bit yang ingin digeser. Nilai tersebut dikonversi menjadi integer.
9. Pergeseran bit ke kiri (<<)<br>
hasil_2013 = angka1_2013 << jumlah_geser_2013<br>
Operator << menggeser bit angka1_2013 ke sebelah kiri sebanyak jumlah_geser_2013 posisi.<br>
Secara sederhana, pergeseran kiri satu posisi biasanya sama dengan mengalikan bilangan dengan 2.<br>
Contoh:<br>
5 = 0101<br>
5 << 1 = 1010 = 10
10. Menampilkan hasil geser kiri<br>
print("\nBitwise geser kiri (<<)")<br>
print(angka1_2013,"<<",jumlah_geser_2013,"=",hasil_2013)<br>
print("Biner hasil =",bin(hasil_2013))<br>
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))<br>
Menampilkan hasil pergeseran dalam bentuk desimal, biner, dan biner 8 bit.
11. Pergeseran bit ke kanan (>>)<br>
hasil_2013 = angka1_2013 >> jumlah_geser_2013<br>
Operator >> menggeser bit angka1_2013 ke sebelah kanan sebanyak jumlah posisi yang ditentukan.<br>
Secara sederhana, untuk bilangan positif, pergeseran kanan satu posisi biasanya sama dengan membagi bilangan dengan 2 dan mengambil bagian bulatnya.<br>
Contoh:<br>
10 = 1010<br>
10 >> 1 = 0101 = 5
12. Menampilkan hasil geser kanan<br>
print("\nBitwise geser kiri (>>)")<br>
print(angka1_2013,">>",jumlah_geser_2013,"=",hasil_2013)<br>
print("Biner hasil =",bin(hasil_2013))<br>
print("Biner hasil (8 bit) =",format(hasil_2013,"08b"))

---
## Tugas Praktikum

**Soal :**

<div align="center">
<img src="image/soal_tugas.png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas3_2611532013
</div>

In [ ]:
print("=== SISTEM TRANSAKSI TOKO ===")
nama_2013 = input("Masukkan Nama Pelanggan : ")
status_2013 = input("Masukkan Status Pelanggan (member/nonmember) : ")
total_2013 = int(input("Masukkan Total Belanja : "))
jumlah_2013 = int(input("Masukkan Jumlah Barang : "))
promo_2013 = input("Masukkan Kode Promo : ")

print("\n=== DATA TRANSAKSI ===")
print(f"Nama Pelanggan       : {nama_2013}")
print(f"Status Pelanggan     : {status_2013}")
print(f"Total Belanja        : Rp{total_2013}")
print(f"Jumlah Barang        : {jumlah_2013}")
print(f"Kode Promo           : {promo_2013}")

kode_promo_2013 = ["HEMAT10", "HEMAT20", "HIDUPMAHASISWA", "HIDUPFTI"] #list kode promo

syarat_total_2013 = total_2013 >= 200000 # apakah memenuhi syarat belanja
syarat_jumlah_2013 = jumlah_2013 >= 3 # apakah memenuhi syarat barang
status_valid_2013 = status_2013 == "member" # apakah user member
promo_valid_2013 = promo_2013 in kode_promo_2013 # apakah kode promo ada dalam list

print("\n=== HASIL VALIDASI ===")
print(f"Belanja >= Rp200000        : {syarat_total_2013}")
print(f"Jumlah Barang >= 3         : {syarat_jumlah_2013}")
print(f"Status Member              : {status_valid_2013}")
print(f"Kode Promo Tersedia        : {promo_valid_2013}")
print(f"Mendapatkan Diskon         : {syarat_jumlah_2013 or syarat_total_2013}")
print(f"Mendapatkan Promo          : {promo_valid_2013}")

diskon_2013 = 0.05

print("\n=== HASIL PERHITUNGAN ===")
print(f"Diskon                      : Rp{diskon_2013 * total_2013}")
print(f"Total Pembayaran            : Rp{total_2013 - total_2013 * diskon_2013}")
print(f"Rata-rata Harga Barang      : Rp{(total_2013 - total_2013 * diskon_2013)/jumlah_2013}")

print("\n=== HAK AKSES PELANGGAN ===")
print(f"Kode Hak Akses              : ...")
print(f"Member Access               : {status_valid_2013}")
print(f"Promo Access                : {promo_valid_2013}")
print(f"Free Shipping Access        : ...")

kode_transaksi_2013 = int(status_valid_2013) << 0 | int(syarat_total_2013) << 1 | int(syarat_jumlah_2013) << 2 | int(promo_valid_2013) << 3
kode_referensi_2013 = int(status_valid_2013) << 0 | int(syarat_total_2013) << 1 | int(promo_valid_2013) << 3

print("\n=== OPERASI BITWISE ===")
print("=== Kode Status Transaksi ===")
print(f"{format(int(status_valid_2013) << 0,"04b")} | {format(int(syarat_total_2013) << 1,"04b")} | {format(int(syarat_jumlah_2013) << 2,"04b")} | {format(int(promo_valid_2013) << 3,"04b")}")
print(f"Kode Biner   : {format(kode_transaksi_2013,"04b")}")
print(f"Kode Desimal : {kode_transaksi_2013}")

print("\n=== Pemeriksaan Status ===")
print("Cek Member")
print(f"{format(kode_transaksi_2013,"04b")} & {format(int(status_valid_2013) << 0,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_2013) & int(status_valid_2013) << 0,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_2013) & int(status_valid_2013) << 0}")

print("Cek Promo")
print(f"{format(kode_transaksi_2013,"04b")} & {format(int(promo_valid_2013) << 3,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_2013) & int(promo_valid_2013) << 3,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_2013) & int(promo_valid_2013) << 3}")

print("\n=== Perbandingan Status ===")
print(f"Kode Transaksi : {format(kode_transaksi_2013,"04b")}")
print(f"Kode Referensi : {format(kode_referensi_2013,"04b")}")
print(f"{format(kode_transaksi_2013,"04b")} ^ {format(kode_referensi_2013,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_2013) ^ (kode_referensi_2013),"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_2013) ^ (kode_referensi_2013)}")

print("\n=== Shift ===")
print(f"{format(kode_transaksi_2013,"04b")} << 1")
print(f"Hasil Biner   : {format((kode_transaksi_2013) << 1,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_2013) << 1}") 
print("=== SELESAI ===")

=== SISTEM TRANSAKSI TOKO ===

=== DATA TRANSAKSI ===
Nama Pelanggan       : adib
Status Pelanggan     : member
Total Belanja        : Rp250000
Jumlah Barang        : 4
Kode Promo           : HEMAT10

=== HASIL VALIDASI ===
Belanja >= Rp200000        : True
Jumlah Barang >= 3         : True
Status Member              : True
Kode Promo Tersedia        : True
Mendapatkan Diskon         : True
Mendapatkan Promo          : True

=== HASIL PERHITUNGAN ===
Diskon                      : Rp12500.0
Total Pembayaran            : Rp237500.0
Rata-rata Harga Barang      : Rp59375.0

=== HAK AKSES PELANGGAN ===
Kode Hak Akses              : ...
Member Access               : True
Promo Access                : True
Free Shipping Access        : ...

=== OPERASI BITWISE ===
=== Kode Status Transaksi ===
0001 | 0010 | 0100 | 1000
Kode Biner   : 1111
Kode Desimal : 15

=== Pemeriksaan Status ===
Cek Member
1111 & 0001
Hasil Biner   : 0001
Hasil Desimal : 1
Cek Promo
1111 & 1000
Hasil Biner   : 1000
Hasil

<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

**Penjelasan :** 
1. Menampilkan judul program<br>
print("=== SISTEM TRANSAKSI TOKO ===")<br>
Menampilkan judul utama program.
2. Memasukkan data pelanggan<br>
nama_2013 = input("Masukkan Nama Pelanggan : ")<br>
status_2013 = input("Masukkan Status Pelanggan (member/nonmember) : ")<br>
total_2013 = int(input("Masukkan Total Belanja : "))<br>
jumlah_2013 = int(input("Masukkan Jumlah Barang : "))<br>
promo_2013 = input("Masukkan Kode Promo : ")<br>
Program meminta nama, status pelanggan, total belanja, jumlah barang, dan kode promo. int() digunakan untuk mengubah total belanja dan jumlah barang menjadi integer.
3. Menampilkan data transaksi<br>
print(f"Nama Pelanggan       : {nama_2013}")<br>
print(f"Status Pelanggan     : {status_2013}")<br>
print(f"Total Belanja        : Rp{total_2013}")<br>
print(f"Jumlah Barang        : {jumlah_2013}")<br>
print(f"Kode Promo           : {promo_2013}")<br>
Menampilkan kembali data yang telah dimasukkan. f-string digunakan agar nilai variabel dapat langsung dimasukkan ke dalam teks.
4. Menentukan kode promo yang tersedia<br>
kode_promo_2013 = ["HEMAT10", "HEMAT20", "HIDUPMAHASISWA", "HIDUPFTI"]<br>
Membuat sebuah list yang berisi kode-kode promo yang tersedia.
5. Mengecek syarat total belanja<br>
syarat_total_2013 = total_2013 >= 200000<br>
Mengecek apakah total belanja minimal Rp200.000. Hasilnya berupa True atau False.
6. Mengecek jumlah barang<br>
syarat_jumlah_2013 = jumlah_2013 >= 3<br>
Mengecek apakah pelanggan membeli minimal 3 barang.
7. Mengecek status member<br>
status_valid_2013 = status_2013 == "member"<br>
Mengecek apakah status pelanggan sama dengan "member". Jika sama persis, hasilnya True.
8. Mengecek kode promo<br>
promo_valid_2013 = promo_2013 in kode_promo_2013<br>
Operator in digunakan untuk mengecek apakah kode promo yang dimasukkan terdapat di dalam list kode_promo_2013.
9. Menampilkan hasil validasi<br>
print(f"Belanja >= Rp200000        : {syarat_total_2013}")<br>
print(f"Jumlah Barang >= 3         : {syarat_jumlah_2013}")<br>
print(f"Status Member              : {status_valid_2013}")<br>
print(f"Kode Promo Tersedia        : {promo_valid_2013}")<br>
Menampilkan hasil dari setiap pemeriksaan dalam bentuk True atau False.
10. Menentukan mendapatkan diskon<br>
print(f"Mendapatkan Diskon         : {syarat_jumlah_2013 or syarat_total_2013}")<br>
Operator or digunakan. Pelanggan dianggap mendapatkan diskon jika jumlah barang minimal 3 atau total belanja minimal Rp200.000.
11. Menentukan mendapatkan promo<br>
print(f"Mendapatkan Promo          : {promo_valid_2013}")<br>
Menampilkan apakah kode promo yang dimasukkan tersedia.
12. Menentukan nilai diskon<br>
diskon_2013 = 0.05<br>
Variabel diskon_2013 diberi nilai 0.05, yang berarti 5%.
13. Menghitung jumlah diskon<br>
print(f"Diskon : Rp{diskon_2013 * total_2013}")<br>
Menghitung nilai diskon dengan mengalikan total belanja dengan 5%.<br>
Contoh jika belanja Rp200.000:<br>
0.05 × 200000 = Rp10.000
14. Menghitung total pembayaran<br>
print(f"Total Pembayaran : Rp{total_2013 - total_2013 * diskon_2013}")<br>
Mengurangi total belanja dengan jumlah diskon untuk mendapatkan total yang harus dibayar.
15. Menghitung rata-rata harga barang<br>
print(f"Rata-rata Harga Barang : Rp{(total_2013 - total_2013 * diskon_2013)/jumlah_2013}")<br>
Menghitung rata-rata harga setiap barang setelah diskon dengan membagi total pembayaran dengan jumlah barang.
16. Menampilkan hak akses pelanggan<br>
print(f"Member Access : {status_valid_2013}")<br>
print(f"Promo Access : {promo_valid_2013}")<br>
Menampilkan hak akses berdasarkan status member dan validitas kode promo.<br>
Jika pelanggan adalah member, Member Access bernilai True. Jika kode promo tersedia, Promo Access bernilai True.
17. Membuat kode transaksi<br>
kode_transaksi_2013 = int(status_valid_2013) << 0 | int(syarat_total_2013) << 1 | int(syarat_jumlah_2013) << 2 | int<br>(promo_valid_2013) << 3
Bagian ini mengubah nilai Boolean menjadi angka dan menyusunnya menggunakan operasi bitwise.<br>
True dikonversi menjadi 1, sedangkan False menjadi 0.<br>
Setiap kondisi ditempatkan pada posisi bit berbeda:<br>
Status member → bit 0<br>
Total belanja → bit 1<br>
Jumlah barang → bit 2<br>
Promo → bit 3<br>
Operator << digunakan untuk menggeser bit, sedangkan | digunakan untuk menggabungkan semua bit tersebut.
18. Membuat kode referensi<br>
kode_referensi_2013 = int(status_valid_2013) << 0 | int(syarat_total_2013) << 1 | int(promo_valid_2013) << 3<br>
Membuat kode referensi dengan menggunakan status member, total belanja, dan promo. Kondisi jumlah barang tidak dimasukkan ke dalam kode ini.
19. Menampilkan kode biner setiap kondisi<br>
print(f"{format(int(status_valid_2013) << 0,"04b")} | {format(int(syarat_total_2013) << 1,"04b")} | {format(int<br>(syarat_jumlah_2013) << 2,"04b")} | {format(int(promo_valid_2013) << 3,"04b")}")
Menampilkan masing-masing kondisi dalam bentuk biner 4 bit.<br>
Fungsi:<br>
format(nilai, "04b")<br>
digunakan untuk mengubah nilai menjadi biner dengan panjang minimal 4 bit.
20. Menampilkan kode transaksi<br>
print(f"Kode Biner   : {format(kode_transaksi_2013,"04b")}")<br>
print(f"Kode Desimal : {kode_transaksi_2013}")<br>
Menampilkan kode transaksi dalam bentuk biner dan desimal.
21. Pemeriksaan status member<br>
print(f"{format(kode_transaksi_2013,"04b")} & {format(int(status_valid_2013) << 0,"04b")}")<br>
Menampilkan operasi bitwise AND antara kode transaksi dan bit status member.<br>
print(f"Hasil Biner   : {format((kode_transaksi_2013) & int(status_valid_2013) << 0,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_2013) & int(status_valid_2013) << 0}")<br>
Menghasilkan nilai berdasarkan pemeriksaan bit status member.
22. Pemeriksaan promo<br>
print(f"{format(kode_transaksi_2013,"04b")} & {format(int(promo_valid_2013) << 3,"04b")}")<br>
Menampilkan operasi AND untuk memeriksa bit promo.<br>
print(f"Hasil Biner   : {format((kode_transaksi_2013) & int(promo_valid_2013) << 3,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_2013) & int(promo_valid_2013) << 3}")<br>
Menampilkan hasil pemeriksaan promo dalam bentuk biner dan desimal.
23. Menampilkan kode transaksi dan referensi<br>
print(f"Kode Transaksi : {format(kode_transaksi_2013,"04b")}")<br>
print(f"Kode Referensi : {format(kode_referensi_2013,"04b")}")<br>
Menampilkan kedua kode dalam bentuk biner 4 bit untuk dibandingkan.
24. Membandingkan kode dengan XOR<br>
print(f"{format(kode_transaksi_2013,"04b")} ^ {format(kode_referensi_2013,"04b")}")<br>
Operator ^ digunakan untuk membandingkan bit dari kode transaksi dan kode referensi. Bit yang berbeda akan menghasilkan 1.<br>
print(f"Hasil Biner   : {format((kode_transaksi_2013) ^ (kode_referensi_2013),"04b")}")<br>
Baris ini menghitung XOR antara kode_transaksi_2013 dan kode_referensi_2013
25. Pergeseran kode transaksi<br>
print(f"{format(kode_transaksi_2013,"04b")} << 1")<br>
Menampilkan operasi pergeseran bit kode transaksi sebanyak 1 posisi ke kiri.<br>
print(f"Hasil Biner   : {format((kode_transaksi_2013) << 1,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_2013) << 1}")<br>
Menggeser seluruh bit kode transaksi satu posisi ke kiri dan menampilkan hasilnya dalam bentuk biner dan desimal.
26. Menampilkan tanda selesai<br>
print("=== SELESAI ===")<br>
Menampilkan pesan bahwa program telah selesai dijalankan.

</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB III<br>KESIMPULAN</h1>

Berdasarkan praktikum yang telah dilakukan, dapat disimpulkan bahwa Python memiliki berbagai jenis operator yang digunakan untuk menjalankan operasi dalam program. Operator tersebut meliputi operator aritmatika, perbandingan, logika, penugasan, bitwise, dan lainnya. Setiap operator memiliki fungsi yang berbeda dan dapat digunakan sesuai kebutuhan program.

Dengan memahami berbagai operasi tersebut, mahasiswa dapat membuat program sederhana yang mampu melakukan perhitungan, membandingkan nilai, memeriksa kondisi, serta melakukan pemeriksaan terhadap data. Pemahaman mengenai operator juga menjadi salah satu dasar penting dalam mempelajari pemrograman Python lebih lanjut.
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">DAFTAR PUSTAKA</h1>

1. Python Software Foundation. (2026). Python 3 Documentation. Python Software Foundation. https://docs.python.org/3/

2. Python Software Foundation. (2026). The Python Tutorial. Python Software Foundation. https://docs.python.org/3/tutorial/

3. W3Schools. (2026). Python Operators. W3Schools. https://www.w3schools.com/python/python_operators.asp

4. GeeksforGeeks. (2026). Python Operators. GeeksforGeeks. https://www.geeksforgeeks.org/python/python-operators/
</div>